### PR TITLE
Stylo: Correct style match for element instances under a use-element …

### PR DESCRIPTION
**Pull request by CJKu**
This patch is the last one of bug 265894.
It fixes two things:
1. Do not cross shadow tree boundary while matching rules.
2. Change display value to 'inline' while cloning a root symbol element instance.

Bugzilla link:
https://bugzilla.mozilla.org/show_bug.cgi?id=265894
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17427)

### CODE REVIEW

**Comment by CJKu** [comment]
@heycam

File: components/style/gecko/wrapper.rs [hunk 1]
```diff
@@ -1656,6 +1657,24 @@ impl<'le> ::selectors::Element for GeckoElement<'le> {
...
+        if self.flags() & (NODE_IS_ANONYMOUS_ROOT as u32) == 0 {
+            return false
+        }
+
+        match self.parent_element() {
+            Some(e) => {
+                // If this element is the shadow root of an use-element shadow
+                // tree, according to the spec, we should not match rules
+                // cross the shadow DOM boundary.
+                e.get_local_name().as_ptr() == atom!("use").as_ptr() &&
```
Comment (emilio) [inline components/style/gecko/wrapper.rs]: nit: Use `*e.get_local_name() == local_name!("use")` instead, and `*e.get_namespace() == ns!("http://...")`

Comment (CJKu) [inline components/style/gecko/wrapper.rs]: e is a GeckoElement, get_local_name returns &WeakAtom, and type of &local_name!("use") is &Atom, so we will get a compile error like this:
 1669 |                 e.get_local_name() == &local_name!("use")
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::cmp::PartialEq<gecko_string_cache::Atom>` is not implemented for `gecko_string_cache::WeakAtom`

Comment (emilio) [inline components/style/gecko/wrapper.rs]: You can dereference the `Atom` then:

```rust
e.get_local_name == &**local_name!("use")
```

or something like that.

**Comment by emilio** [comment]
@bors-servo r+

Stylo builds are orange on autoland and I believe this should fix it.

**Comment by emilio** [comment]
I'll fix the namespace/localname comparison as soon as this lands.

**Comment by emilio** [comment]
@bors-servo p=10

 * orange

**Pull request merged**